detectorCommon/src/xtalk_neighbors_moduleThetaMergedSegmentation.cpp

```diff
@@ -0,0 +1,245 @@
+#include "detectorCommon/DetUtils_k4geo.h"
+#include "detectorCommon/xtalk_neighbors_moduleThetaMergedSegmentation.h"
+
+// DD4hep
+#include "DDG4/Geant4Mapping.h"
+#include "DDG4/Geant4VolumeManager.h"
+
+// Geant
+#include "G4NavigationHistory.hh"
+
+// ROOT
+#include "TGeoBBox.h"
+
+#ifdef HAVE_GEANT4_UNITS
+#define MM_2_CM 1.0
+#else
+#define MM_2_CM 0.1
+#endif
+
+#include <iostream>
+
+#include <unordered_set>
+
+namespace det {
+namespace xtalk {
+
+// use it for module-theta merged readout (FCCSWGridModuleThetaMerged_k4geo)
+std::vector<std::pair<uint64_t, double>> xtalk_neighbours_ModuleThetaMerged(const dd4hep::DDSegmentation::FCCSWGridModuleThetaMerged_k4geo& aSeg,
+						   const dd4hep::DDSegmentation::BitFieldCoder& aDecoder,
+						   const std::vector<std::string>& aFieldNames,
+						   const std::vector<std::vector<std::pair<int, int>>>& aFieldExtremes_layer,
+						   uint64_t aCellId) {
+
+  std::vector<std::pair<uint64_t, double>> xtalk_neighbours;
+  
+  // check that field names and extremes have the proper length
+  if (aFieldNames.size() != 3 || aFieldExtremes_layer[0].size()!=3) {
+    std::cout << "ERROR: the vectors aFieldNames and aFieldSizes should be of length = 3, corresponding to the theta/module/layer fields" << std::endl;
+    std::cout << "ERROR: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }  
+  // find index of layer, module and theta in the field name vector
+  int idModuleField(-1);
+  int idThetaField(-1);
+  int idLayerField(-1);
+  for (int itField = 0; itField < aFieldNames.size(); itField++) {
+    if (aFieldNames[itField] == aSeg.fieldNameModule())
+      idModuleField = itField;
+    else if (aFieldNames[itField] == aSeg.fieldNameTheta())
+      idThetaField = itField;
+    else if (aFieldNames[itField] == aSeg.fieldNameLayer())
+      idLayerField = itField;
+   
+  }
+  if (idModuleField < 0) {
+    std::cout << "WARNING: module field " << aSeg.fieldNameModule() << " not found in aFieldNames vector" << std::endl;
+    std::cout << "WARNING: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }
+  if (idThetaField < 0) {
+    std::cout << "WARNING: theta field " << aSeg.fieldNameTheta() << " not found in aFieldNames vector" << std::endl;
+    std::cout << "WARNING: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }
+  if (idLayerField < 0) {
+    std::cout << "WARNING: layer field " << aSeg.fieldNameLayer() << " not found in aFieldNames vector" << std::endl;
+    std::cout << "WARNING: will return empty neighbour map" << std::endl;
+    return xtalk_neighbours;
+  }
+  
+  double dummy_xtalk=0;
+  double dummy_xtalk_radial=0.35; // type 1
```

Comment:
As their names indicate, these number used to be dummy numbers for crosstalk coefficients. I have updated them from Juska's latest measurement of crosstalks, apart from "type 4", which is still set to a reasonable guess:
https://indico.cern.ch/event/1368231/contributions/5904291/attachments/2834580/4953190/pcb_update_and_v2_juska_v0.pdf
In the next round of development, the crosstalk coefficients will become configurable, rather than fixed numbers in the code.